src/reactionsystem.jl

```diff
@@ -617,70 +628,77 @@ function ReactionSystem(eqs, iv, unknowns, ps;
                         continuous_events = nothing,
                         discrete_events = nothing,
                         metadata = nothing)
-                        
-    name === nothing &&
+               
```

Comment:
When I did this tehre was some functions in reactionssyetm.jl that I had to go through line by line to figure out what was going on. While I did this I took the time to add comments what was going on, and to fix some weird formating. I only touched functions that I needed to go through for this PR, the remaining ones I left for another time.